polkadot/xcm/xcm-executor/src/lib.rs

```diff
@@ -624,6 +624,20 @@ impl<Config: config::Config> XcmExecutor<Config> {
 				Ok(())
 			},
 			DepositReserveAsset { assets, dest, xcm } => {
+				// we need to do this take/put cycle to solve wildcards and get exact assets to be
+				// weighed
+				let to_weigh = self.holding.saturating_take(assets.clone());
+				self.holding.subsume_assets(to_weigh.clone());
+
+				let mut message_to_weigh =
+					vec![ReserveAssetDeposited(to_weigh.into()), ClearOrigin];
+				message_to_weigh.extend(xcm.0.clone().into_iter());
+				let (_, fee) =
+					validate_send::<Config::XcmSender>(dest.clone(), Xcm(message_to_weigh))?;
```

Comment:
should we just modify send to not charge fee and have another method that charges send fee and call it here? so that we don't need to calculate the send fee multiple times